Lib/test/test_tempfile.py

```diff
@@ -1288,6 +1288,18 @@ def test_writelines(self):
         buf = f.read()
         self.assertEqual(buf, b'xyz')
 
+    def test_writelines_rollover(self):
+        # Verify writelines rolls over before exhausting the iterator
+        f = self.do_create(max_size=2)
+
+        def it():
+            yield b'xy'
+            self.assertFalse(f._rolled)
+            yield b'z'
+            self.assertTrue(f._rolled)
+
+        f.writelines(it())
```

Comment:
```suggestion
        f.writelines(it())
        pos = f.seek(0)
        self.assertEqual(pos, 0)
        buf = f.read()
        self.assertEqual(buf, b'xyz')
```

Reply:
I opted against this, as this is checked by the test immediately above this one, but I can add it.

Reply:
The test above uses `max_size=0` which doesn't test the new code you added. I think it's safer to add this.